ring/ring_test.go

```diff
@@ -189,12 +201,9 @@ func TestDoBatchZeroInstances(t *testing.T) {
 	cleanup := func() {
 	}
 	desc := NewDesc()
-	r := Ring{
-		cfg:      Config{},
-		ringDesc: desc,
-		strategy: NewDefaultReplicationStrategy(),
-	}
-	require.Error(t, DoBatch(ctx, Write, &r, keys, callback, cleanup))
+	r := newRingForTesting(Config{}, false)
+	r.setRingStateFromDesc(desc, false, false, true)
```

Comment:
Why true for `updateReadOnlyInstances`?

Reply:
That seems unnecessary indeed, and not what previous code was doing. I'll remove it.